File: src/main/java/skywolf46/extrautility/util/ItemCounter.java
package skywolf46.extrautility.util;

import lombok.AllArgsConstructor;
import org.bukkit.inventory.ItemStack;

@AllArgsConstructor
public class ItemCounter {
    private ItemStack item;

    public ItemStack getItem() {
        return item;
    }

    public int getAmount(){
        return item.getAmount();
    }

    public void setAmount(int am) {
        if (am < 0)
            item = null;
        else
            item.setAmount(am);
    }
}

File: src/main/kotlin/skywolf46/extrautility/util/ItemCounter.kt
package skywolf46.extrautility.util

import lombok.AllArgsConstructor
import org.bukkit.inventory.ItemStack


@AllArgsConstructor
class ItemCounter {
    var item: ItemStack? = null
        private set
    var amount: Int
        get() = item!!.amount
        set(am) {
            if (am < 0) item = null else item!!.amount = am
        }
}